napari/utils/_injection/_processors.py

```diff
@@ -70,11 +70,11 @@ def _add_layer_data_tuples_to_viewer(
         data = data if isinstance(data, list) else [data]
         for datum in ensure_list_of_layer_data_tuple(data):
             # then try to update a viewer layer with the same name.
-            if len(datum) > 1 and (name := datum[1].get("name")):
```

Comment:
I guess to be backwards compatible it's as simple as updating `ensure_list_of_layer_data_tuple` so that actually returns a `LayerDataTuple`?

Reply:
> However, right now, using dot notation (.attributes...) breaks backward compatibility.

yeah, this is critical.  And similar to my other comment.  Whatever conveniences you'd like to provide with a named tuple, we _must_ allow plugins to not use it, so we need to be extremely careful to either _always_ cast to a named tuple internally (which I could take or leave really), or simply keep the current logic of assuming a plain tuple